### PR TITLE
Add Flying Carpet to FixedLayer

### LINKED ISSUE #3812: Fix Viewability for amp-fx-flying-carpet component
The problem with the current amp-fx-flying-carpet is that the viewability reported is that of the container and not the ad inside the container. 
CC @jridgewell @kashyapnitin

### PR DESCRIPTION
**Pull request by jridgewell**
The final final fix for #3812, so that resources correctly detects the contents of a Flying Carpet are fixed position. That triggers #5624's behavior, and that gives us the correct visibility metrics! Huray!

Also adds type checking.

Fixes #3812.
/cc @jasti, @muxin

### CODE REVIEW

File: extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js [hunk 1]
```diff
@@ -148,11 +163,11 @@ class AmpFlyingCarpet extends AMP.BaseElement {
...
     return nodes.filter(node => {
-      if (node.nodeType === /* Element */1) {
+      if (node.nodeType === /* Element */ 1) {
         return true;
       }
 
-      if (node.nodeType === /* Text */3) {
+      if (node.nodeType === /* Text */ 3) {
         // Is there a non-whitespace character?
         return /\S/.test(node.textContent);
```
Comment (aghassemi) [inline extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js]: unrelated to this PR but `return node.textContent.trim() !== '';` will be significantly faster and less CPU-intensive than the regex

Comment (aghassemi) [inline extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js]: unrelated to this PR but return node.textContent.trim() !== ''; will be significantly faster and less CPU-intensive than the regex

Comment (jridgewell) [inline extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js]: Jsperf's back up. 😉

Comment (jridgewell) [inline extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js]: https://jsperf.com/text-node-regex-vs-textcontent-trim

Comment (aghassemi) [inline extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js]: In a pure trim vs regex test, trim is 70% faster: https://jsperf.com/regexvtrim

Comment (jridgewell) [inline extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js]: I see your 70% and raise you the common case: https://jsperf.com/jridgewell-regex-v-trim

Comment (jridgewell) [inline extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js]: Or, getting rid of the regex allocation: https://jsperf.com/jridgewell-regex-v-trim-2

Comment (aghassemi) [inline extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js]: lol, problem with regex is that its CPU intensive. Both of your tests, when run on actual phone, are still 30%-50% slower (sure on desktop difference in negligible)

**Pull request merged**